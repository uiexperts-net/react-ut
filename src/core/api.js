
export const fetchUser = async () => {
    const response = await fetch('http://test.com/api/user');
    const data = await response.json();
    return data;
}

export const fetchUserWithCB = async (cb) => {
    const response = await fetch('http://test.com/api/user');    
    if(!response.ok){
        return cb({data:null,isSuccess:false})
    }
    const data = await response.json();
    return cb(data,true);
}
