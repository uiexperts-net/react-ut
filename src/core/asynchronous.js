
const fetchUserAPI = async () => {
    const response = await fetch('http://localhost/api/user');
    const data = await response.json();
    return data;
}

const fetchUser = () => {
    return new Promise((resolve, reject) => {
        // Simulating an async response for fetch User
        setTimeout(() => {
            const success = true; // Simulated success status

            if (success) {
                // Resolve the promise if the task is successful
                resolve({ name: 'TestUser', contactNo: 4503430342 });
            } else {
                // Reject the promise if the task fails
                reject("Error: Unable to fetch data");
            }
        }, 2000); // Simulating a delay of 2 seconds
    });
}

export const processUserData = async () => {
    const userData = await fetchUser()
    return userData;
}

export const processUserDataWithAPI = async () => {
    const userData = await fetchUserAPI()
    return userData;
}

export const processUserDataWithCallback = (cbFunc) => {    
    setTimeout(() => {
        cbFunc('SUCCESS', { name: 'TestUser', contactNo: 4503430342 })
    }, 1000);
    fetchUser()
}