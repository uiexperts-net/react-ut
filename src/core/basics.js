
export default getValues = ()=>[1,2,3,4,5]

export const addValues = (a,b)=> (a+b);

export const subtractValues = (a,b)=> (a-b);


