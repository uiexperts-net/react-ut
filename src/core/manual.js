

export const getManualData = ()=>'Maual Data';