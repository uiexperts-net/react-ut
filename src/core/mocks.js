import axios from 'axios';

export const fetchUserList = ()=>{
    return axios.get('/users').then(response=>response.data);
}


export const processUserList = (userList, formatUser) => {
    const formattedUsers = userList.map(user => formatUser(user));
    return formattedUsers;
}