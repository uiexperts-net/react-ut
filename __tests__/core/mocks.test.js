import { fetchUserList, processUserList } from "../../src/core/mocks";

import axios from "axios";
import { addValues,subtractValues } from "../../src/core/basics";

jest.mock('axios');
const userList = ['selvam', 'ravi', 'ramjith']

jest.mock('../../src/core/basics',()=>{
    const originalModule = jest.requireActual('../../src/core/basics');
    return {
        __esModule: true,
        ...originalModule,
        addValues: jest.fn((a,b)=>(a+b+100))
    }
})

describe('Demonstration of using Mocks for Unit Testing', () => {
    
    test('validate mock function', () => {
        const mockFormatUserCallback = jest.fn((userName) => `MR.${userName.toUpperCase()}`)
        processUserList(userList, mockFormatUserCallback);
        expect(mockFormatUserCallback.mock.calls).toHaveLength(3)
        expect(mockFormatUserCallback.mock.calls[1][0]).toBe('ravi')
        expect(mockFormatUserCallback.mock.results[1].value).toBe('MR.RAVI')
    })

    test('validate mock function return values', () => {
        const mockCallbackReturn = jest.fn()
        mockCallbackReturn.mockReturnValueOnce('one').mockReturnValueOnce('two').mockReturnValue('default');
        processUserList(userList, mockCallbackReturn);
        expect(mockCallbackReturn.mock.results[1].value).toBe('two')
    })

    test('validate mock modules', () => {
        axios.get.mockResolvedValue({ data: userList });
        return fetchUserList().then(data => expect(data).toEqual(userList))
    })

    test('validate partial mocking of a module',()=>{    
            expect(addValues(2,3)).toBe(105);
            expect(subtractValues(5,2)).toBe(3);            
            expect(subtractValues(5,2)).toMatchSnapshot();            

    })

});