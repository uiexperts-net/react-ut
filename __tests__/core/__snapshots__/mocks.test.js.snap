// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demonstration of using Mocks for Unit Testing validate partial mocking of a module 1`] = `3`;
